src/Icicle/Source/Checker/Constraint.hs

```diff
@@ -174,7 +174,8 @@ generateQ qq@(Query (c:_) _)
             retKey     <- Temporality TemporalityElement . TypeVar <$> fresh
             retVal     <- Temporality TemporalityElement . TypeVar <$> fresh
             (q',sq,t') <- withBind k retKey
-                        $ withBind v retVal rest
+                        $ withBind v retVal
+                        $ removeElementBinds >> rest
```

Comment:
hmm this looks to me like it would add "k : Element K" and then remove it straight after?